server/services/frontendComponentService.ts

```diff
@@ -0,0 +1,17 @@
+import type { FrontendComponent } from 'frontendComponentApiClient'
+import { FrontendComponentApiClient } from '../data'
+import logger from '../../logger'
+
+export default class FrontendComponentService {
+  constructor(private readonly frontendApiClient: FrontendComponentApiClient) {}
+
+  async getComponents(component: 'header' | 'footer', userToken: string): Promise<FrontendComponent> {
+    try {
+      const frontendComponentResponse = await this.frontendApiClient.getComponents(component, userToken)
+      return frontendComponentResponse
+    } catch (error) {
+      logger.error(`Error retrieving DPS frontend components: ${error}`)
```

Comment:
Here's the catch block I was referring to in my previous comment - I think this is correct
(though I think there might be a problem with the type itself potentially not having a `problemRetrievingData` field? -  but I'll look into it after you've merged )